ietf/ietfauth/views.py

```diff
@@ -440,15 +441,27 @@ def confirm_new_email(request, auth):
     form = NewEmailForm({ "new_email": email })
     can_confirm = form.is_valid() and email
     new_email_obj = None
+    created = False
     if request.method == 'POST' and can_confirm and request.POST.get("action") == "confirm":
-        new_email_obj = Email.objects.create(address=email, person=person, origin=username)
+        try: 
+            new_email_obj, created = Email.objects.get_or_create(
+                address=email, 
+                person=person, 
+                defaults={'origin': username},
+            )
+        except IntegrityError:
+            can_confirm = False
+            form.add_error(
+                None, "Email address is in use by another user. Please contact the secretariat for assistance."
+            )
```

Comment:
Wouldn't this again leak the existence of email addresses?

Reply:
Only if you can receive messages sent to that address - this code runs in response to following the confirmation link that was just emailed there.

Further, the duplicate confirmation link will only be generated if you attempt to add the address between the times its rightful owner adds it and confirms it.

My thought was that if you can engineer your way through that you probably already know the address is there. I think this is more likely to come up for someone having account issues so it is better to nudge them to get help. I could make it treat this like an invalid link, though, and that would leak the least possible information.

Reply:
You could also get to this code by brute force posts, but it would require creating a valid signature over the username and email, and _that_ essentially requires knowing the site's secret key. If that's compromised, there are worse problems.